cpp/src/arrow/compute/exec/asof_join_node_test.cc

```diff
@@ -92,6 +95,12 @@ Result<BatchesWithSchema> MakeBatchesFromNumString(
         ARROW_ASSIGN_OR_RAISE(Datum as_string, Cast(num_batch.values[i], utf8()));
         ARROW_ASSIGN_OR_RAISE(Datum as_type, Cast(as_string, type));
         values.push_back(as_type);
+      } else if (Type::BOOL == type->id()) {
+        ARROW_ASSIGN_OR_RAISE(Datum div_two, Divide(num_batch.values[i], two));
```

Comment:
Can you add some comments here? It's a bit confusing to use various math/cast operation to create a boolean value from the original value. Also, what is the original value that is used to create the boolean value? (string? int?)

Reply:
Sure, I'll add. The original value is the `int64` value noted [here](https://github.com/westonpace/arrow/pull/24#discussion_r1013038707).

Reply:
Sounds good - I think this starts to make more sense to be but when I looked at the code I got pretty confused XD